psi4/driver/procrouting/proc.py

```diff
@@ -1333,6 +1333,10 @@ def scf_helper(name, post_scf=True, **kwargs):
 
     if cast:
         core.print_out("\n  Computing basis projection from %s to %s\n\n" % (ref_wfn.basisset().name(), base_wfn.basisset().name()))
+        if ref_wfn.basisset().n_ecp_core() != base_wfn.basisset().n_ecp_core():
+            raise ValidationError("Projecting from basis ({}) with ({}) ECP electrons to basis ({}) with ({}) ECP electrons will be a disaster. Select a compatible cast-up basis with `set guess_basis YOUR_BASIS_HERE`.".format(
+                                  ref_wfn.basisset().name(), ref_wfn.basisset().n_ecp_core(), base_wfn.basisset().name(), base_wfn.basisset().n_ecp_core()))
+
```

Comment:
Yes. Even better would be to check if the ECPs match, but that's for sure a smaller problem.